From a37616a6f5ddbac2f075b8c5342b0fb984e2a446 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Thu, 12 Aug 2010 09:30:58 +0100
Patch-mainline: 2.6.37
References: fate#310916
Git-commit: 6ba3ddd9838f5e4d6ac7c6dce95648d205e11bff
Subject: [PATCH 1304/2588] drm/i915/overlay: Missing breaks between case
 statements for color depth

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
(cherry picked from commit 6ba3ddd9838f5e4d6ac7c6dce95648d205e11bff)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_overlay.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_overlay.c b/drivers/gpu/drm/i915/intel_overlay.c
index 5ed1783..4972b5c 100644
--- a/drivers/gpu/drm/i915/intel_overlay.c
+++ b/drivers/gpu/drm/i915/intel_overlay.c
@@ -675,10 +675,13 @@ static void update_colorkey(struct intel_overlay *overlay,
 			    struct overlay_registers *regs)
 {
 	u32 key = overlay->color_key;
+
 	switch (overlay->crtc->base.fb->bits_per_pixel) {
 	case 8:
 		regs->DCLRKV = 0;
 		regs->DCLRKM = CLK_RGB8I_MASK | DST_KEY_ENABLE;
+		break;
+
 	case 16:
 		if (overlay->crtc->base.fb->depth == 15) {
 			regs->DCLRKV = RGB15_TO_COLORKEY(key);
@@ -687,10 +690,13 @@ static void update_colorkey(struct intel_overlay *overlay,
 			regs->DCLRKV = RGB16_TO_COLORKEY(key);
 			regs->DCLRKM = CLK_RGB16_MASK | DST_KEY_ENABLE;
 		}
+		break;
+
 	case 24:
 	case 32:
 		regs->DCLRKV = key;
 		regs->DCLRKM = CLK_RGB24_MASK | DST_KEY_ENABLE;
+		break;
 	}
 }
 
-- 
1.7.6

